src/nvim/edit.c

```diff
@@ -1488,8 +1488,8 @@ void edit_putchar(int c, int highlight)
         int fix_col = mb_fix_col(pc_col, pc_row);
 
         if (fix_col != pc_col) {
-          screen_putchar(' ', pc_row, fix_col, attr);
-          --curwin->w_wcol;
+          grid_putchar(&default_grid, ' ', pc_row, fix_col, attr);
```

Comment:
I think we can use `wp->w_grid` here (and all other changed calls in this file), and remove `w_winrow`/`w_wincol`) and the new fallback logic should handle the not-multigrid case.

Reply:
I'm not sure if I should be removing `w_winrow` from this: https://github.com/UtkarshMe/neovim/blob/6313b02704e7ac357911e169c71881f5b43fcd9d/src/nvim/edit.c#L559
`curwin->w_wrow` is relative to the window i.e. `w_winrow`. I don't know what this line does.

Also, I should be changing `w_height`/`w_width` to `w_grid.Rows`/`w_grid.Columns` too, right?